impeller/display_list/dl_aiks_canvas.cc

```diff
@@ -0,0 +1,1204 @@
+  canvas_.Rotate(Degrees{degrees});
+}
+
+// |flutter::DlCanvas|
+void DlAiksCanvas::Skew(SkScalar sx, SkScalar sy) {
+  canvas_.Skew(sx, sy);
+}
+
+// |flutter::DlCanvas|
+void DlAiksCanvas::Transform2DAffine(SkScalar mxx,
+                                     SkScalar mxy,
+                                     SkScalar mxt,
+                                     SkScalar myx,
+                                     SkScalar myy,
+                                     SkScalar myt) {
+  // clang-format off
+  TransformFullPerspective(
+    mxx, mxy,  0, mxt,
+    myx, myy,  0, myt,
+    0  ,   0,  1,   0,
+    0  ,   0,  0,   1
+  );
+  // clang-format on
+}
+
+// |flutter::DlCanvas|
+void DlAiksCanvas::TransformFullPerspective(SkScalar mxx,
+                                            SkScalar mxy,
+                                            SkScalar mxz,
+                                            SkScalar mxt,
+                                            SkScalar myx,
+                                            SkScalar myy,
+                                            SkScalar myz,
+                                            SkScalar myt,
+                                            SkScalar mzx,
+                                            SkScalar mzy,
+                                            SkScalar mzz,
+                                            SkScalar mzt,
+                                            SkScalar mwx,
+                                            SkScalar mwy,
+                                            SkScalar mwz,
+                                            SkScalar mwt) {
+  // The order of arguments is row-major but Impeller matrices are
+  // column-major.
+  // clang-format off
+  auto xformation = Matrix{
+    mxx, myx, mzx, mwx,
+    mxy, myy, mzy, mwy,
+    mxz, myz, mzz, mwz,
+    mxt, myt, mzt, mwt
+  };
+  // clang-format on
+  canvas_.Transform(xformation);
+}
+
+// |flutter::DlCanvas|
+void DlAiksCanvas::TransformReset() {
+  canvas_.ResetTransform();
+}
+
+// |flutter::DlCanvas|
+void DlAiksCanvas::Transform(const SkMatrix* matrix) {
+  if (!matrix) {
+    return;
+  }
+  canvas_.Transform(ToMatrix(*matrix));
+}
+
+void DlAiksCanvas::Transform(const SkM44* matrix44) {
+  if (!matrix44) {
+    return;
+  }
+  canvas_.Transform(ToMatrix(*matrix44));
+}
+
+static Entity::ClipOperation ToClipOperation(
+    flutter::DlCanvas::ClipOp clip_op) {
+  switch (clip_op) {
+    case flutter::DlCanvas::ClipOp::kDifference:
+      return Entity::ClipOperation::kDifference;
+    case flutter::DlCanvas::ClipOp::kIntersect:
+      return Entity::ClipOperation::kIntersect;
+  }
+}
+
+// |flutter::DlCanvas|
+void DlAiksCanvas::ClipRect(const SkRect& rect, ClipOp clip_op, bool is_aa) {
+  canvas_.ClipRect(skia_conversions::ToRect(rect), ToClipOperation(clip_op));
+}
+
+// |flutter::DlCanvas|
+void DlAiksCanvas::ClipRRect(const SkRRect& rrect, ClipOp clip_op, bool is_aa) {
+  if (rrect.isSimple()) {
```

Comment:
Simple allows different radx and rady. When it says "equal radii" it means all 4 corners are equal, not that the horizontal and vertical radii are the same.

Also, it has an isRect() that could be trapped and converted into a ClipRect()

Reply:
This was copy pasted from dl_dispatcher

Reply:
Filed: https://github.com/flutter/flutter/issues/133793